diesel/src/r2d2.rs

```diff
@@ -2,6 +2,89 @@
 //!
 //! Note: This module requires enabling the `r2d2` feature
 //!
+//! # Example
+//!
+//! The below snippet is a contrived example emulating a web application,
+//! where one would first initialize the pool in the `main()` function
+//! (at the start of a long-running process). One would then pass this
+//! pool struct around as shared state, which, here, we've emulated using
+//! threads instead of routes.
+//!
+//! ```rust
+//! # include!("doctest_setup.rs");
+//! use diesel::result::Error;
+//! use diesel::pg::PgConnection;
+//! use diesel::prelude::*;
+//! use diesel::r2d2::ConnectionManager;
+//! use diesel::r2d2::Pool;
+//! use std::env;
+//! use std::thread;
+//!
+//! pub type PostgresPool = Pool<ConnectionManager<PgConnection>>;
```

Comment:
```suggestion
//! pub type DatabasePool = Pool<ConnectionManager<DbConnection>>;
```

+ Defining `type DbConnection = PgConnection/MysqlConnection/…` in `doctest_setup.rs` should allow you to run this test for all backends.